src/runtimes/node/bundler.js

```diff
@@ -36,20 +39,39 @@ const bundleJsFile = async function ({
 }) {
   // De-duping external and ignored modules.
   const external = [...new Set([...externalModules, ...ignoredModules])]
-  const jsFilename = `${basename(destFilename, extname(destFilename))}.js`
-  const bundlePath = join(destFolder, jsFilename)
+
+  // To be populated by the native modules plugin with the names, versions and
+  // paths of any Node modules with native dependencies.
   const nativeNodeModules = {}
+
+  // To be populated by the dynamic imports plugin with the names of the Node
+  // modules that include imports with dynamic expressions.
   const nodeModulesWithDynamicImports = new Set()
+
+  // To be populated by the dynamic imports plugin with any paths (in a glob
+  // format) to be included in the bundle in order to make a dynamic import
+  // work at runtime.
+  const dynamicImportsIncludedPaths = new Set()
+
+  // The list of esbuild plugins to enable for this build.
   const plugins = [
     getNativeModulesPlugin(nativeNodeModules),
-    getDynamicImportsPlugin({ moduleNames: nodeModulesWithDynamicImports, srcDir }),
+    getDynamicImportsPlugin({
+      basePath,
+      includedPaths: dynamicImportsIncludedPaths,
+      moduleNames: nodeModulesWithDynamicImports,
+      srcDir,
+    }),
   ]
+
+  // The version of ECMAScript to use as the build target. This will determine
+  // whether certain features are transpiled down or left untransformed.
   const nodeTarget = getBundlerTarget(config.nodeVersion)
 
-  // esbuild will format `sources` relative to the sourcemap file, which is a
-  // sibling of `bundlePath`. We use `sourceRoot` to establish that relation.
-  // They are URLs, so even on Windows they should use forward slashes.
-  const sourceRoot = dirname(bundlePath).replace(/\\/g, '/')
+  // esbuild will format `sources` relative to the sourcemap file, which lives
+  // in `destFolder`. We use `sourceRoot` to establish that relation. They are
+  // URLs, not paths, so even on Windows they should use forward slashes.
+  const sourceRoot = destFolder.replace(/\\/g, '/')
```

Comment:
`bundlePath` was being computed with `join(destFolder, jsFilename)`, so it doesn't make sense to be getting its dirname here. We can use `destFolder` directly.